compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs

```diff
@@ -399,16 +395,18 @@ fn codegen_float_intrinsic_call<'tcx>(
         | sym::ceilf64
         | sym::truncf32
         | sym::truncf64
-        | sym::nearbyintf32
-        | sym::nearbyintf64
+        | sym::round_ties_even_f32
+        | sym::round_ties_even_f64
         | sym::sqrtf32
         | sym::sqrtf64 => {
             let val = match intrinsic {
                 sym::fabsf32 | sym::fabsf64 => fx.bcx.ins().fabs(args[0]),
                 sym::floorf32 | sym::floorf64 => fx.bcx.ins().floor(args[0]),
                 sym::ceilf32 | sym::ceilf64 => fx.bcx.ins().ceil(args[0]),
                 sym::truncf32 | sym::truncf64 => fx.bcx.ins().trunc(args[0]),
-                sym::nearbyintf32 | sym::nearbyintf64 => fx.bcx.ins().nearest(args[0]),
+                sym::round_ties_even_f32 | sym::round_ties_even_f64 => {
+                    fx.bcx.ins().nearest(args[0])
+                }
```

Comment:
This means the cranelift backend will actually change behavior with this PR -- the `round_ties_even` functions will lower to the "nearest" instruction rather than to `rint` libcalls. @bjorn3 is that okay?

I am also surprised to not see a case for `round` here (rounding away from zero); is that the one rounding mode cranelit does not support natively?

Reply:
If this ever lowers to `nearbyint` then we do not currently provide that in libm so there would be problems on targets without a system `libm`. I don't expect this to be a problem for cranelift though.